File: code/leetcode/3.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        max_len = 0
        r = 0
        stri = []

        for x, y in enumerate(s):
            stri.append(y)
            while len(stri) != len(set(stri)):
                stri.pop(0)
            if len(stri) > max_len:
                max_len = len(stri)

        return max_len

File: code/cpp/leetcode/3.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int freq[256]={0};
        int left=0;
        int maxLen=0;
        for(int right=0;right<s.length();right++){
            freq[(unsigned char)s[right]]++;
            while(freq[(unsigned char)s[right]]>1){freq[(unsigned char)s[left]]--; left++;}
            int currLen=right-left+1;
            if(currLen>maxLen){maxLen=currLen;}
        }
        return maxLen;
    }
};